cmd/wstail/tail.go

```diff
@@ -13,135 +13,112 @@ import (
 	"os"
 	"os/exec"
 	"path/filepath"
+	"strconv"
 	"strings"
-	"syscall"
-	"time"
 
 	log "github.com/Sirupsen/logrus"
-	"github.com/gorilla/websocket"
 	"github.com/gravitational/trace"
 )
 
 const defaultTailSource = "/var/log/messages"
 
+// defaultTailLimit defines how many last lines will tail output by default
+const defaultTailLimit = 100
+
 // maxDumpLen defines the maximum size of the string window to output
 // to console after a failed interpretation (as in failure to decode JSON)
 const maxDumpLen = 128
 
-// tailMaxDepth defines how many last lines will tail output with no filter set
-const tailMaxDepth = 100
-
 // rotatedLogUncompressed names the first uncompressed (potentially in use)
 // rotated log file as named by savelog
 const rotatedLogUncompressed = "messages.0"
 
-// hard limit for log file size which is 1.5Mb
-const tailFileSizeLimit = 1.5 * 1024 * 1024
-
-func tailer(ws *websocket.Conn, filter filter) {
-	matcher := buildMatcher(filter)
-	log.Infof("active filter: %v (%v)", filter, matcher)
-	defer ws.Close()
-
+func logReader(filePath string, filter filter, limit string) ([]byte, error) {
 	dir := filepath.Dir(defaultTailSource)
 	names, err := readDir(dir)
 	if err != nil {
-		log.Errorf("failed to read log directory: %v", err)
-		return
+		return nil, trace.Wrap(err, "failed to read log directory: %v")
+	}
+
+	_, err = os.Stat(filePath)
```

Comment:
I thought that `trace.ConvertSystemError(err)` will make descriptive error message